src/main/java/com/bmc/truesight/apiclient/client/TrueSightAPIClient.java

```
@@ -0,0 +1,126 @@
+     
+    /**
+     * addMeasures() invoking the API /v1/measurements.
+     * returns a TrueSightClientResponse on successful execution.
+     * The method takes a list of Measurement Objects (1 or more), a TrueSightCredentials object
+     * and the sleep time in milliseconds that the system need to wait during a retry.
+     * 
+     * @param measures List of Measurement objects which need to be pushed to TrueSight.
+     * @param credentials   Basic authentication credentials required for REST API call.
+     * @param retrySleepTime Number of milliseconds the client has to wait during a retry.
+     * @return TrueSightClientResponse  Provides HTTP status code and JSON response from the API.
+     * @throws TrueSightClientException Wraps other exceptions thrown during the method execution.
+     * @see TrueSightClientResponse
+     * @see Measurement
+     * @see TrueSightCredentials
+     */
+    public TrueSightClientResponse addMeasures(List<Measurement> measures, TrueSightCredentials credentials, int retrySleepTime) throws TrueSightClientException {
+        
+        LOG.info("Inovking addMeasures() ...");
+        TrueSightClientResponse result = null;
+        if(measures != null && measures.size() > 0) 
+        {
+            LOG.debug("getMeasures() called with {} Measurements", measures.size());
+            List<Object> bulkMeasures = TrueSightAPIClientUtility.toBulkEntryMeasureFormat(measures);            
+            String endpoint = TrueSightAPIClientUtility.getHost() + Configuration.MEASURE_ENDPOINT;
+            result = TrueSightAPIClientUtility.executePost(bulkMeasures, credentials, endpoint, retrySleepTime);
+            LOG.debug("ClientResponse: {}", result.toString());
+        }
+        else
+        {
+            LOG.error("Failed to add Measurement: Measurement list is null or empty");
+            throw new TrueSightClientException("Failed to add Measurement: Measurement list is null or empty");
```

Should we throw an exception when the measure object sent in was `null` or `empty`? I don't think so. We should probably `warn` and leave it there.